piker/pp.py

```diff
@@ -276,6 +277,62 @@ def lifo_update(
 
         return new_size, self.be_price
 
+    def calc_be_price(self) -> float:
+
+        size: float = 0
+        cb_tot_size: float = 0
+        cost_basis: float = 0
+        be_price: float = 0
+
+        for tid, entry in self.clears.items():
+            clear_size = entry['size']
+            clear_price = entry['price']
+            new_size = size + clear_size
+
+            # old size minus the new size gives us size diff with
+            # +ve -> increase in pp size
+            # -ve -> decrease in pp size
+            size_diff = abs(new_size) - abs(size)
+
+            if new_size == 0:
+                cost_basis = 0
+                cb_tot_size = 0
+                be_price = 0
+
+            elif size_diff > 0:
+                # only an increaze in size of the position contributes
+                # the breakeven price, a decrease does not.
+
+                cost_basis += (
+                    # weighted price per unit of
+                    clear_price * abs(clear_size)
+                    +
+                    # transaction cost
+                    (copysign(1, new_size) * entry['cost'] * 2)
```

Comment:
the `* 2` bit should really be changed to allow user override as well as other modelling methods for exit cost estimation.

For an example `kraken` provides their fee tiers in the [asset info endpoint](https://docs.kraken.com/rest/#tag/Market-Data/operation/getAssetInfo), so it's definitely possible to dynamically calculate a breakeven level depending on the exit strategy.